### PR TITLE
Add WaveMode configuration key in SOAR Goodman spectrograph

### PR DESCRIPTION
**Pull request by Allonck**
In the ```configurations keys``` and ```raw_header_cards``` were added the WAVMODE Header key. This is useful to see when dealing with typical configurations like 400_M1 and 400_M2. 

Also in ```meta.py``` was added a new variable ```filter2```. The main reason of the latter is because goodman uses two filters in their wheel and others spectrographs can benefit if they need the keyword.

### CODE REVIEW

**Review by rcooke-ast — COMMENTED** [review]
Thanks, @Allonck - much appreciated!

I'm running the dev-suite tests on your branch now, and will approve once the tests pass.

**Comment by rcooke-ast** [comment]
The dev-suite checks out, so I'm therefore approving! Thanks very much for your contribution, @Allonck!

image

**Review by rcooke-ast — APPROVED** [review]
dev-suite passes.

**Comment by rcooke-ast** [comment]
One thing, @Allonck - I've redirected this PR to the `develop` branch, and updated the branch. I hope this is OK with you. If you wish to make additional changes, make sure you pull these changes first.

**Review by kbwestfall — CHANGES_REQUESTED** [review]
Thanks, @Allonck !  I changed the PR to go into `pypeit:develop` instead of `pypeit:release`.  Can you also add a brief comment in `doc/releases/1.15.1.rst` file that summarizes these changes under the "Instrument-specific" section?

**Comment by Allonck** [comment]
@kbwestfall Thank you for your feedback and guidance. I've made the requested changes and added a brief comment in the ``doc/releases/1.15.1dev.rst`` file under the "Instrument-specific" section, summarizing the updates as per your suggestion. 

Please feel free to review the changes at your convenience. Let me know if there's anything else I can assist you with.

File: doc/releases/1.15.1dev.rst [hunk 1]
```diff
@@ -10,15 +10,23 @@ Dependency Changes
...
+- Allow the selection of a secondary filter option  for all spectrographs
+that need it; see new ``filter2`` parameter.
+=======
 - When dealing with `bkg_redux` reduction (i.e., A-B), a sky model (called `bkg_redux_skymodel`)
   is generated using the A frame, while the existing `skymodel` is generated using the A-B frame.
   This allows to have an actual 1D extracted sky spectrum and to perform the flexure correction for
   `bkg_redux` reduction.
 - Various improvements in the flexure correction and added the possibility to use a modeled archive
   sky spectrum generated with `pypeit.wavemodel.nearIR_modelsky()` to perform the flexure correction.
+>>>>>>> 5dbd2b1a78748a2d947b8c2e60b9e0de4543589d
```
Comment (kbwestfall) [inline doc/releases/1.15.1dev.rst]: It looks like you had a merge conflict.  Can you please remove lines 13, 16, and 23?

**Review by kbwestfall — CHANGES_REQUESTED** [review]
Thanks, @Allonck !  Nearly ready to merge.  Just one final request.

File: doc/releases/1.15.1dev.rst (hunk 1 above)
Comment (Allonck) [reply doc/releases/1.15.1dev.rst]: Hi @kbwestfall 

Thank you for your continued guidance and feedback. I've addressed the merge conflict as per your instructions and removed the lines 13, 16, and 23.

Please let me know if everything looks good now or if there's anything else you need from my end.

Appreciate your help in getting this PR ready for merging!

**Review by kbwestfall — APPROVED** [review]
Thanks for the quick fix!  Approving.  I'll merge once the tests pass.

**Comment by kbwestfall** [comment]
Hi @Allonck .  Sorry for the delay on this.  Can you please merge the upstream pypeit `release` branch into this branch of your fork?

**Pull request merged**